### PR TITLE
Add ability to customise fonts in siteconfig

### PR DESCRIPTION
**Pull request by maxime-rainville**
Originally created by @bergice.

# Parent issue
* https://github.com/silverstripeltd/open-sourcerers/issues/82

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "name": "silverstripe/theme-fontpicker",
+    "description": "Customise site fonts from the admin settings",
+    "type": "silverstripe-vendormodule",
+    "keywords": [
+        "silverstripe"
+    ],
+    "license": "BSD-3-Clause",
+    "require": {
+        "silverstripe/framework": "^4.1",
```
Comment (maxime-rainville) [inline composer.json]: It would make more sense to target a specific version of admin and/or siteconfig.

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Fontpicker Module
+
+[![Build Status](https://travis-ci.org/silverstripe/fontpicker.svg?branch=master)](https://travis-ci.org/silverstripe/fontpicker)
```
Comment (maxime-rainville) [inline readme.md]: These URL need updating

File: src/Extensions/FontpickerSiteConfigExtension.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                        'Main font family'
+                    ),
+                    $fonts
+                )
+            ]
+        );
+    }
+
+    /**
+     * If HeaderBackground is not set, assume no theme colours exist and populate some defaults if the colour
```
Comment (maxime-rainville) [inline src/Extensions/FontpickerSiteConfigExtension.php:63]: Comment doesn't seem to make sense in this context.

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        "silverstripe/fontpicker": "*"
```
Comment (maxime-rainville) [inline composer.json]: When we have a proper release of theme, this needs to be updated. Could we set it to something like `^0.0.x-dev`

**Review by maxime-rainville — COMMENTED** [review]
Can we make sure the Travis build is configured correctly?

Can we add a very basic test?

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    },
+    "require-dev": {
+        "phpunit/phpunit": "^5.7",
+        "squizlabs/php_codesniffer": "^3"
+    },
+    "autoload": {
+        "psr-4": {
+            "SilverStripe\\Fontpicker\\": "src/"
```
Comment (maxime-rainville) [inline composer.json]: Wrong namespace.

Add an autoload namespace for tests as well.

File: src/Extensions/FontpickerSiteConfigExtension.php [hunk 6]
```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace SilverStripe\Fontpicker\Extensions;
```
Comment (maxime-rainville) [inline src/Extensions/FontpickerSiteConfigExtension.php]: Wrong namespace.

**Review by maxime-rainville — COMMENTED** [review]
The colour picker theme package has an enabled setting, but this one doesn't.

As mentioned on the other PR, I think if you install this package, it should be enabled by default.

https://github.com/silverstripe/silverstripe-theme-colorpicker/pull/1/files#r303283358

**Comment by chillu** [comment]
I don't want to create a `1` branch before we actually alias `master` to `2.x`. And that should only happen once we introduce breaking changes. Until then, it's just more complexity to maintain two parallel branches. Maybe that was just a temporary fix to get peer review going? After merging this, we should reset `1` to `master` and delete the `1` branch.

**Comment by chillu** [comment]
Closing this as "merged" (it's already in `master`, we shouldn't be using the `1` branch)

**Pull request closed without merging**